tools/crm_mon.c

```diff
@@ -2009,10 +2010,30 @@ mon_refresh_display(gpointer user_data)
         out->reset(out);
     }
 
-    rc = pcmk__output_cluster_status(out, st, cib, current_cib, fence_history,
-                                     show, show_opts, options.only_node,
-                                     options.only_rsc, options.neg_location_prefix,
-                                     output_format == mon_output_monitor);
+    if ((cib != NULL) && (cib->state != cib_disconnected)) {
```

Comment:
what happens now when this isn't true? i thought that maybe we would still try to show fence history and pacemakerd status

Reply:
It doesn't care about the CIB state now. `pcmk__output_cluster_status()` uses `current_cib`. As far as my testing went, `current_cib` stayed populated. Only `cib_connect()` and `pacemakerd_status()` query the CIB and alter `current_cib`.

I also wasn't able to hit `mon_refresh_display()` after a CIB disconnection until I added this patch. Right now, `mon_refresh_display()` only gets called **before** `mon_cib_connection_destroy()`.

If `current_cib` were `NULL`, presumably yes something down the call chain from `pcmk__output_cluster_status()` would end up printing the things we do have access to (fence history, etc.).

Not to say with certainty that anything can't happen... it gets tough to follow.

Reply:
Test the original code to see if the other status is still shown correctly (running pacemaker-fenced in standalone mode should be sufficient for this purpose). If not, then we can forget about it, otherwise we'll need to do this in pcmk__output_cluster_status() and the "cluster-status" message.

Reply:
If `crm_mon` can't connect to `pacemakerd` (e.g., because the fencer is running in standalone mode), it doesn't show anything except
```
Could not connect to pacemakerd: Connection refused
crm_mon: Connection to cluster failed: Connection refused
```
in one-shot mode, or `Waiting until cluster is available on this node ...` in console mode. And nothing at all in daemonized mode.

If I run `pacemakerd`, that's going to start all the subdaemons. If I then kill `pacemaker-based`, `pacemakerd` restarts `pacemaker-based`, `pacemaker-fenced`, and `pacemaker-controld`. In the brief interim period when daemons are being restarted, the node state is updated to `OFFLINE`, the resource state disappears, and the current DC disappears.

Reply:
OK, this should be fine then